src/SimplePromiseQueue.ts

```diff
@@ -1,25 +1,26 @@
 // tslint:disable: variable-name
 export default class SimplePromiseQueue {
   private readonly _queue: Array<() => Promise<void>> = []
-  private _flushing = false
+  private _runner: Promise<void> | null = null
 
   public enqueue(task: () => Promise<void>) {
     this._queue.push(task)
-    if (!this._flushing) { return this.flushQueue() }
-    return Promise.resolve()
+    this.flushQueue()
   }
 
-  public flushQueue() {
-    this._flushing = true
-
-    const chain = (): Promise<void> | void => {
-      const nextTask = this._queue.shift()
-      if (nextTask) {
-        return nextTask().then(chain)
-      } else {
-        this._flushing = false
+  public async flushQueue(): Promise<void> {
+    const runner = async () => {
+      while (this._queue.length) {
+        const nextTask = this._queue.shift()!
+        await nextTask()
       }
+      this._runner = null
+    }
```

Comment:
Might be cleaner to make this a method on the class instead of an anonymous function.

Reply:
This was intentional. I didn't want anyone else calling this function because it would break the queue if there are multiple runners.

Reply:
I agree though, it'd be nice to define this as a separate function. The old code did this by setting a variable while the queue is flushing, but it's not enough for our use case because we want to save the promise as well (which feels impossible to do if the promise is the result of the currently running function).